packages/test-sync/src/commonTest/kotlin/io/realm/kotlin/test/mongodb/common/AppTests.kt

```diff
@@ -438,12 +438,20 @@ class AppTests {
 
             // Create a configuration pointing to the metadata Realm for that app
             val metadataDir = "${app.configuration.syncRootDirectory}/mongodb-realm/${app.configuration.appId}/server-utility/metadata/"
+
+            // Workaround for https://github.com/realm/realm-core/issues/7876
+            // We cannot validate if the test app metadata realm is encrypted directly, as it is cached
+            // and subsequent access wont validate the encryption key. Copying the Realm allows to bypass
+            // the cache.
+            PlatformUtils.copyFile(metadataDir + "sync_metadata.realm", metadataDir + "copy_sync_metadata.realm")
+
             val wrongKey = TestHelper.getRandomKey()
             val config = RealmConfiguration
                 .Builder(setOf())
-                .name("sync_metadata.realm")
+                .name("copy_sync_metadata.realm")
                 .directory(metadataDir)
                 .encryptionKey(wrongKey)
+                .schemaVersion(7)
```

Comment:
Why do we need to specify the schema version? the test is validating the encryption key...

Reply:
Because the schema version that the client is using to write the metadata realm is greater than 0 which is used by default. If we don't set it we will get a migration error when trying to open the realm.